To: vim_dev@googlegroups.com
Subject: Patch 7.3.242
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.242
Problem:    Illegal memory access in after_pathsep().
Solution:   Check that the pointer is not at the start of the file name.
	    (Dominique Pelle)
Files:	    src/misc2.c


*** ../vim-7.3.241/src/misc2.c	2011-07-07 16:20:45.000000000 +0200
--- src/misc2.c	2011-07-07 17:05:41.000000000 +0200
***************
*** 3247,3253 ****
  #if defined(FEAT_MBYTE) || defined(PROTO)
  /*
   * Return TRUE if "p" points to just after a path separator.
!  * Take care of multi-byte characters.
   * "b" must point to the start of the file name
   */
      int
--- 3247,3253 ----
  #if defined(FEAT_MBYTE) || defined(PROTO)
  /*
   * Return TRUE if "p" points to just after a path separator.
!  * Takes care of multi-byte characters.
   * "b" must point to the start of the file name
   */
      int
***************
*** 3255,3261 ****
      char_u	*b;
      char_u	*p;
  {
!     return vim_ispathsep(p[-1])
  			     && (!has_mbyte || (*mb_head_off)(b, p - 1) == 0);
  }
  #endif
--- 3255,3261 ----
      char_u	*b;
      char_u	*p;
  {
!     return p > b && vim_ispathsep(p[-1])
  			     && (!has_mbyte || (*mb_head_off)(b, p - 1) == 0);
  }
  #endif
*** ../vim-7.3.241/src/version.c	2011-07-07 16:44:33.000000000 +0200
--- src/version.c	2011-07-07 17:05:49.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     242,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
260. Co-workers have to E-mail you about the fire alarm to get
     you out of the building.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
